spanner/cloud-client/src/main/java/com/example/spanner/LoadCsvExample.java

```diff
@@ -0,0 +1,274 @@
+      if (!tableColumns.containsKey(csvHeader)) {
+        throw new IllegalArgumentException(
+            "File header " + csvHeader + " does not match any database table column name.");
+      }
+    }
+    return true;
+  }
+
+  /** Initialize CSV Parser format based on user specified option flags **/
+  public static CSVFormat setFormat(CommandLine cmd) {
+    CSVFormat parseFormat;
+    // Set file format type
+    if (cmd.hasOption("f")) {
+      switch (cmd.getOptionValue("f").toUpperCase()) {
+        case EXCEL:
+          parseFormat = CSVFormat.EXCEL;
+          break;
+        case POSTGRESQL_CSV:
+          parseFormat = CSVFormat.POSTGRESQL_CSV;
+          break;
+        case POSTGRESQL_TEXT:
+          parseFormat = CSVFormat.POSTGRESQL_TEXT;
+          break;
+        default:
+          parseFormat = CSVFormat.DEFAULT;
+      }
+    } else {
+      parseFormat = CSVFormat.DEFAULT;
+    }
+    // Set null string representation
+    if (cmd.hasOption("n")) {
+      parseFormat = parseFormat.withNullString(cmd.getOptionValue("n"));
+    }
+    // Set delimiter character
+    if (cmd.hasOption("d")) {
+      if (cmd.getOptionValue("d").length() != 1) {
+        throw new IllegalArgumentException("Invalid delimiter character entered.");
+      }
+      parseFormat = parseFormat.withDelimiter(cmd.getOptionValue("d").charAt(0));
+    }
+    // Set escape character
+    if (cmd.hasOption("e")) {
+      if (cmd.getOptionValue("e").length() != 1) {
+        throw new IllegalArgumentException("Invalid escape character entered.");
+      }
+      parseFormat = parseFormat.withEscape(cmd.getOptionValue("e").charAt(0));
+    }
+    // Set parser to parse first row as headers
+    if (cmd.hasOption("h") && cmd.getOptionValue("h").equalsIgnoreCase("True")) {
+      parseFormat = parseFormat.withFirstRecordAsHeader();
+      hasHeader = true;
+    }
+    return parseFormat;
+  }
+
+  /** Verifies that if file has a header, that the record is mapped to a column header name
+   * and that the record itself is not null **/
+  public static boolean validHeaderField(CSVRecord record, String columnName) {
+    return hasHeader && record.isMapped(columnName) && record.get(columnName) != null;
+  }
+
+  /** Verifies that if the file has no header, that the record at the given index is not null **/
+  public static boolean validNonHeaderField(CSVRecord record, int index) {
+    return !hasHeader && record.get(index) != null;
+  }
+
+  /** Write CSV file data to Spanner using JDBC Mutation API **/
+  public static void writeToSpanner(Iterable<CSVRecord> records, String tableName, CommandLine cmd)
+      throws SQLException {
+    System.out.println("Writing data into table...");
+    List<Mutation> mutations = new ArrayList<>();
+    for (CSVRecord record : records) {
+      int index = 0;
+      WriteBuilder builder = Mutation.newInsertOrUpdateBuilder(tableName);
+      for (String columnName : tableColumns.keySet()) {
```

Comment:
Optional:
Consider adding a comment here about column ordering and why it's important that tableColumns is a LinkedHashMap.